services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentServiceITest.kt

```diff
@@ -0,0 +1,215 @@
+import fi.hel.haitaton.hanke.attachment.HANKE_TUNNUS
+import fi.hel.haitaton.hanke.attachment.USERNAME
+import fi.hel.haitaton.hanke.attachment.common.ApplicationAttachmentRepository
+import fi.hel.haitaton.hanke.attachment.common.ApplicationAttachmentType
+import fi.hel.haitaton.hanke.attachment.common.ApplicationAttachmentType.MUU
+import fi.hel.haitaton.hanke.attachment.common.ApplicationAttachmentType.VALTAKIRJA
+import fi.hel.haitaton.hanke.attachment.common.AttachmentNotFoundException
+import fi.hel.haitaton.hanke.attachment.common.AttachmentScanStatus
+import fi.hel.haitaton.hanke.attachment.common.AttachmentScanStatus.OK
+import fi.hel.haitaton.hanke.attachment.common.AttachmentUploadException
+import fi.hel.haitaton.hanke.attachment.testFile
+import fi.hel.haitaton.hanke.factory.AlluDataFactory
+import java.util.*
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.EnumSource
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.MediaType.TEXT_HTML_VALUE
+import org.springframework.security.test.context.support.WithMockUser
+import org.springframework.test.context.ActiveProfiles
+import org.testcontainers.junit.jupiter.Testcontainers
+
+@Testcontainers
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("default")
+@WithMockUser(USERNAME)
+class ApplicationAttachmentServiceITest : DatabaseTest() {
+    @Autowired private lateinit var applicationAttachmentService: ApplicationAttachmentService
+    @Autowired private lateinit var applicationAttachmentRepository: ApplicationAttachmentRepository
+    @Autowired private lateinit var alluDataFactory: AlluDataFactory
+    @Autowired private lateinit var hankeRepository: HankeRepository
+
+    @Test
+    fun `getMetadataList should return related metadata list`() {
+        val application = initApplication()
+        (1..2).forEach { _ ->
+            applicationAttachmentService.addAttachment(
+                applicationId = application.id!!,
+                attachmentType = MUU,
+                attachment = testFile()
+            )
+        }
+
+        val result = applicationAttachmentService.getMetadataList(application.id!!)
+
+        assertThat(result).hasSize(2)
+        assertThat(result).each { d ->
+            d.transform { it.id }.isNotNull()
+            d.transform { it.fileName }.endsWith("file.pdf")
+            d.transform { it.createdByUserId }.isEqualTo(USERNAME)
+            d.transform { it.createdAt }.isNotNull()
+            d.transform { it.scanStatus }.isEqualTo(OK)
+            d.transform { it.applicationId }.isEqualTo(APPLICATION_ID)
+            d.transform { it.attachmentType }.isEqualTo(MUU)
+        }
+    }
+
+    @Test
+    fun `getContent when status is OK should succeed`() {
```

Comment:
Yes, one test is enough.

In the test, it might be useful for there to be an attachment under another application with the requested ID. So we know a user can't access another application's attachments by changing the id in the request path.

Reply:
Test added. Added test for hanke too.